frontend/lib/resolution/Resolver.cpp

```diff
@@ -4529,22 +4530,23 @@ resolveZipExpression(Resolver& rv, const IndexableLoop* loop, const Zip* zip) {
     // Resolve the leader iterator.
     auto dt = resolveIterDetails(rv, leader, leader, {}, m);
 
+    eltTypes.push_back(dt.idxType);
+
     // Configure what followers should do using the iterator details.
     if (dt.succeededAt == IterDetails::LEADER_FOLLOWER) {
       followerPolicy = IterDetails::FOLLOWER;
       leaderYieldType = dt.leaderYieldType;
-      eltTypes.push_back(dt.idxType);
     } else if (dt.succeededAt == IterDetails::SERIAL) {
       followerPolicy = IterDetails::SERIAL;
-      eltTypes.push_back(dt.idxType);
     } else {
-      return { QualifiedType::UNKNOWN, ErroneousType::get(context) };
+      ret = { QualifiedType::UNKNOWN, ErroneousType::get(context) };
```

Comment:
not sure how strictly we've been sticking to this, but I think the proper style is to only return and create a new ErroneousType if an error has been emitted. Consider just using a boolean flag instead of using ErroneousType to distinguish from "unknown".

Reply:
I'd like to keep it `ErroneousType` because I think ideally we would emit an error at that point, it's just when I first try it I got some test failures. I'll put a TODO on that branch.